pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -58,7 +58,7 @@ in
 , pkg-configDepends ? [], libraryPkgconfigDepends ? [], executablePkgconfigDepends ? [], testPkgconfigDepends ? [], benchmarkPkgconfigDepends ? []
 , testDepends ? [], testHaskellDepends ? [], testSystemDepends ? [], testFrameworkDepends ? []
 , benchmarkDepends ? [], benchmarkHaskellDepends ? [], benchmarkSystemDepends ? [], benchmarkFrameworkDepends ? []
-, testTarget ? ""
+, testTarget ? "", testFlags ? []
```

Comment:
Since the arguments aren't documented here, I guess you could add a helper function in `haskell.lib` and add documentation and an example for this argument.

Although I don't think that is strictly necessary for this PR to be merged, more of a nice-to-have.

I also don't know bash well enough to work out if the `checkFlagsArray` expression is correct, but it looks like it at least works for your two examples, so it looks good to me!

Reply:
I'm a bit sceptical that it is actually a good thing having all these functions for specific arguments. In some cases they are justified when there is an element of “doing the right thing” to it (like `appendConfigureFlags`, …) and when the overrides are more complicated yet reusable (`generateOptparseApplicativeCompion`, `justStaticExecutables`, ...).

In this case `overrideCabal` is perfectly fine for setting this and my intention is to document this in the manual, I have already documented all existing arguments to `haskellPackages.mkDerivation` (which was apparently needed as I had to check the `cabal2nix` source for some details…) and would add `testFlags` there as well when this is merged.

Reply:
I was thinking that ideally this documentation could go here in `pkgs/development/haskell-modules/generic-builder.nix`, immediately above the place where it is defined.

Although it would be good if we could generate the manual directly from this file.

Reply:
This as well, but I think the actual manual is actually more important at this time, I'm hoping `nixdoc` will get some love in the future to support this use case as well (it kinda does currently, but also kinda doesn't).